src/applications/deploying-addons.md

```diff
@@ -0,0 +1,101 @@
+their applications. The end-goal is to provide a way for users to consume services from brokers and
+have their applications use those services without needing detailed knowledge about how those
+services are created or managed.
+
+As an example, most applications deployed by Workflow need a data store of some kind. `deis addons`
+allows applications to consume services like databases that exist somewhere via common environment
+variables like `DATABASE_URL`.
+
+## Listing Available Add-ons
+
+A user can use `deis addons:installed` to see if they have a database provisioned and what plan it
+is using.
+
+```
+$ deis addons:installed | grep -i postgresql
+$ deis-postgresql:standard
+```
+
+If the application doesn’t yet have a database provisioned, a user can create a new database using
+the CLI.
+
+If a user wishes to see all available addons, they can use `deis addons:list`.
+
+```
+$ deis addons:list
+slug             name
+---------------  ----------------------
+bonsai           Bonsai Elasticsearch
+deis-postgresql  Deis Workflow Postgres
+librato          Librato
+```
+
+## Provisioning the Add-on
+
+Most service brokers offers a variety of plans, usually spread across different tiers of service:
+hobby, standard, premium, and enterprise. For a detailed breakdown on the available plans, check
+the documentation for the applicable service broker to help choose the right service tier for the
+application.
+
+For example, to provision a `standard` plan database:
+
+```
+$ deis addons:create deis-postgresql:standard --app wooden-rowboat
+Creating deis-postgresql:standard... done
+Attaching deis-postgresql:standard to wooden-rowboat... done, v5
+```
+
+Depending on the plan, some services can take some time before it becomes available for use. Use
+`deis addons:wait` to wait for the service to become available.
+
+Once the service has been attached to the application, a DATABASE_URL setting will be available in
+the application's configuration and will contain the URL used to access the newly provisioned
+service.
+
+## Deprovisioning the Add-on
+
+To deprovision a `standard` plan database:
+
+```
+$ deis addons:destroy deis-postgresql:standard
+Destroying deis-postgresql:standard... done
+```
+
+## Attaching the Add-on
+
+Once the add-on is provisioned, it is bound to the user's account and can be bound to any number of
+applications the user has permissions to use.
+
+To attach an add-on to a new or existing application:
+
+```
+$ deis addons:attach deis-postgresql:standard --app wooden-rowboat
+Attaching deis-postgresql:standard to wooden-rowboat... done, v5
+```
+
+## Detaching the Add-on
+
+Similarly, an add-on can be unbound from an application, but still be available to the user for
+future binding to another application.
+
+```
+$ deis addons:detach deis-postgresql:standard --app wooden-rowboat
+Detaching deis-postgresql:standard from wooden-rowboat... done, v6
+```
```

Comment:
So let's think about this in a team-oriented fashion:

 - I have a team (a namespace to store all apps deployed by a team)
 - I have many apps deployed in this namespace
 - I have a few instances running in this namespace

I would expect that, as a user, I can bind an app to the instance as well as unbind apps from the instance.

This is essentially useless for the "instance in the app's namespace" use case though. Again I'll give this some more thought. :)